src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,102 @@
-export const Pagination = () => {};
+import React, { useMemo } from 'react';
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (pageNumber: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const itemsNumbers: number[] = getNumbers(1, total);
+  const pages: number[] = getNumbers(1, Math.ceil(total / perPage));
+
+  const firstPageIndex = (currentPage - 1) * perPage;
+  const lastPageIndex = firstPageIndex + perPage < itemsNumbers.length
+    ? firstPageIndex + perPage
+    : itemsNumbers.length;
+  const currentTableData = useMemo(() => {
+    return itemsNumbers.slice(firstPageIndex, lastPageIndex);
+  }, [currentPage, perPage]);
+
+  const firstPageIsActive = currentPage === 1;
+  const lastPageIsActive = currentPage === pages.length;
+
+  const previousPageHandler = () => {
+    if (!firstPageIsActive) {
+      onPageChange(currentPage - 1);
+    }
+  };
+
+  const nextPageHandler = () => {
+    if (!lastPageIsActive) {
+      onPageChange(currentPage + 1);
+    }
+  };
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={cn('page-item', { disabled: firstPageIsActive })}>
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={firstPageIsActive}
+            onClick={previousPageHandler}
+          >
+            «
+          </a>
+        </li>
+        {getNumbers(1, Math.ceil(total / perPage)).map((pageNumber: number) => (
```

Comment:
`getNumbers(1, Math.ceil(total / perPage))` should be transferred to a separate variable and use the finished result.